indra/llcharacter/llik.h

```diff
@@ -402,25 +403,26 @@ class Joint
 public:
     class Config
     {
-    public:
+      public:
         // local info is in parent-frame
-        bool hasLocalPos() const { return (mFlags & CONFIG_FLAG_LOCAL_POS) > 0; }
-        bool hasLocalRot() const { return (mFlags & CONFIG_FLAG_LOCAL_ROT) > 0; }
-        bool hasLocalScale() const { return (mFlags & CONFIG_FLAG_LOCAL_SCALE) > 0; }
-        bool constraintIsDisabled() const { return (mFlags & CONFIG_FLAG_DISABLE_CONSTRAINT) > 0; }
-        void setLocalPos(const LLVector3& pos);
-        void setLocalRot(const LLQuaternion& rot);
-        void setLocalScale(const LLVector3& scale);
-        void disableConstraint() { mFlags |= CONFIG_FLAG_DISABLE_CONSTRAINT; }
-        const LLVector3& getLocalPos() const { return mLocalPos; }
-        const LLQuaternion& getLocalRot() const { return mLocalRot; }
-        const LLVector3& getLocalScale() const { return mLocalScale; }
+        bool                hasLocalPos() const { return (mFlags & CONFIG_FLAG_LOCAL_POS) > 0; }
+        bool                hasLocalRot() const { return (mFlags & CONFIG_FLAG_LOCAL_ROT) > 0; }
+        bool                hasLocalScale() const { return (mFlags & CONFIG_FLAG_LOCAL_SCALE) > 0; }
+        bool                constraintIsDisabled() const { return (mFlags & CONFIG_FLAG_DISABLE_CONSTRAINT) > 0; }
+        void                setLocalPos(const LLVector3 &pos);
+        void                setLocalRot(const LLQuaternion &rot);
+        void                setLocalScale(const LLVector3 &scale);
+        void                disableConstraint() { mFlags |= CONFIG_FLAG_DISABLE_CONSTRAINT; }
+        const LLVector3    &getLocalPos() const { return mLocalPos; }
+        const LLQuaternion &getLocalRot() const { return mLocalRot; }
+        const LLVector3    &getLocalScale() const { return mLocalScale; }
```

Comment:
fwiw I have come to look askance at methods returning references. In this case it's okay, because if we later have to change one of these to return a computed value, we could change its return type at the same time. I just want to note that a virtual method returning a reference is an anti-pattern: any one subclass, present or future, that later needs to return a computed value is likely to lead to ugly unsafe code.